cluster/addons/podsecuritypolicies/unprivileged-addon.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: extensions/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: system.unprivileged-addon
+  annotations:
+    kubernetes.io/description: 'This policy grants the minimum ammount of
+      privilege necessary to run non-privileged kube-system pods. This policy is
+      not intended for use outside of kube-system, and may include further
+      restrictions in the future.'
+    # TODO: Addons should use the default seccomp profile.
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: '*'
+    # 'runtime/default' is already the default, but must be filled in on the
+    # pod to pass admission.
+    apparmor.security.beta.kubernetes.io/defaultProfileName:  'runtime/default'
+    apparmor.security.beta.kubernetes.io/allowedProfileNames: 'runtime/default'
+  labels:
+    kubernetes.io/cluster-service: 'true'
+    addonmanager.kubernetes.io/mode: Reconcile
+spec:
+  privileged: false
+  allowPrivilegeEscalation: false
+  volumes:
+  - 'emptyDir'
+  - 'configMap'
+  - 'secret'
+  hostNetwork: false
+  hostIPC: false
+  hostPID: false
+  # TODO: The addons using this profile should not run as root.
```

Comment:
this TODO will be hard to fix... do we know which addons wouldn't work if you set RunAsNonRoot?

Reply:
I have a google-internal spreadsheet tracking this, I can work on a public version. My guess is this will be pretty straightforward for most non-privileged pods. There are a couple pods that just need root to write to a host-path logfile that's owned by root, but that should be easy to fix.

I wish Kubernetes managed user IDs as a system resource to try and avoid overlapping users. My approach so far has just been to assign everything non-root to `65534` (which is typically `nobody`).

Reply:
> There are a couple pods that just need root to write to a host-path logfile that's owned by root, but that should be easy to fix.

this doesn't allow hostPath volumes... anything else blocking using MustRunAsNonRoot?